### PR TITLE
Better error checks when creating SpatialViewDefinitions

### PR DESCRIPTION
**Pull request by ramanujam-raman**
Better error checks when creating SpatialViewDefinitions.

### CODE REVIEW

File: core/backend/src/test/integration/HubUtility.ts [hunk 1]
```diff
@@ -348,7 +348,7 @@ export class HubUtility {
 
   /** Push an iModel to the Hub */
   public static async pushIModel(requestContext: AuthorizedClientRequestContext, projectId: string, pathname: string, iModelName?: string, overwrite?: boolean): Promise<GuidString> {
-    assert.isTrue(HubMock.isValid, "Must use HubMock for tests that create iModels");
+    assert.isTrue(HubUtility.allowHubBriefcases || HubMock.isValid, "Must use HubMock for tests that create iModels");
```
Comment (kabentley) [inline core/backend/src/test/integration/HubUtility.ts:351]: Why? We don't want to create any tests that don't use mocks - that's why I added this check. Please remove this.

File: core/backend/src/test/integration/IModelWrite.test.ts [hunk 2]
```diff
@@ -1068,8 +1069,72 @@ describe("IModelWriteTest (#integration)", () => {
     schemaLock = await IModelHost.hubAccess.querySchemaLock({ requestContext: superRequestContext, briefcase: superBriefcaseProps });
     assert.isFalse(schemaLock); // Validate no schema locks held by the hub
 
-    /* Cleanup after test */
     await IModelHost.hubAccess.deleteIModel({ requestContext: managerRequestContext, contextId: projectId, iModelId });
   });
 
+  it("create SpatialViewDefinition and throw errors on bad input (#integration)", async () => {
```
Comment (kabentley) [inline core/backend/src/test/integration/IModelWrite.test.ts:1075]: This test is not an integration test. It should be in standalone tests. See Category.test.ts

**Review by kabentley — CHANGES_REQUESTED** [review]

File: core/backend/src/test/integration/IModelWrite.test.ts (hunk 2 above)
Comment (ramanujam-raman) [reply core/backend/src/test/integration/IModelWrite.test.ts:1075]: Moved test to the new location.

File: core/backend/src/test/integration/HubUtility.ts (hunk 1 above)
Comment (ramanujam-raman) [reply core/backend/src/test/integration/HubUtility.ts:351]: I kept this to just quickly test it with the Hub if we needed to - it seemed quite useful - e.g., used it to push some iModels to the Hub the other day. 

Also, we do have this flag elsewhere?

File: core/backend/src/test/integration/IModelWrite.test.ts (hunk 2 above)
Comment (ramanujam-raman) [reply core/backend/src/test/integration/IModelWrite.test.ts]: I have moved it to the new location now - this is definitely a standalone test and not an integration test.

File: core/backend/src/test/integration/HubUtility.ts (hunk 1 above)
Comment (kabentley) [reply core/backend/src/test/integration/HubUtility.ts:351]: i don't want to encourage people to write tests that write to iMH.

File: core/backend/src/test/standalone/ViewDefinition.test.ts [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import { Guid, Id64, Id64String } from "@bentley/bentleyjs-core";
+import { Matrix3d, Range3d, StandardViewIndex, Transform, YawPitchRollAngles } from "@bentley/geometry-core";
+import { Camera, Code, ColorDef, ElementProps, IModel, IModelError, SpatialViewDefinitionProps, SubCategoryAppearance } from "@bentley/imodeljs-common";
+import { assert } from "chai";
+import { AuthorizedBackendRequestContext, CategorySelector, DictionaryModel, DisplayStyle3d, IModelDb, ModelSelector, SpatialCategory, SpatialViewDefinition, StandaloneDb } from "../../imodeljs-backend";
+import { IModelTestUtils, TestUserType } from "../IModelTestUtils";
+
+export async function createNewModelAndCategory(requestContext: AuthorizedBackendRequestContext, rwIModel: IModelDb, parent?: Id64String) {
+  // Create a new physical model.
+  const [, modelId] = await IModelTestUtils.createAndInsertPhysicalPartitionAndModelAsync(requestContext, rwIModel, IModelTestUtils.getUniqueModelCode(rwIModel, "newPhysicalModel"), true, parent);
```
Comment (kabentley) [inline core/backend/src/test/standalone/ViewDefinition.test.ts:14]: use sync function. Get rid of requestContext.

File: core/backend/src/test/standalone/ViewDefinition.test.ts [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      guid: Guid.createValue(),
+    });
+  });
+
+  after(() => {
+    iModel.close();
+  });
+
+  it("create SpatialViewDefinition and throw errors on bad input", async () => {
+    const requestContext = await IModelTestUtils.getUserContext(TestUserType.SuperManager);
```
Comment (kabentley) [inline core/backend/src/test/standalone/ViewDefinition.test.ts:44]: not necessary

File: core/backend/src/test/standalone/ViewDefinition.test.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+      classFullName: "BisCore:SpatialViewDefinition",
+      cameraOn: false,
+      origin: rotation.multiplyTransposeXYZ(rotatedRange.low.x, rotatedRange.low.y, rotatedRange.low.z),
+      extents: rotatedRange.diagonal(),
+      angles,
+      camera: new Camera(),
+    };
+
+    // Bad way to create - error checks
+    assert.throws(() => iModel.elements.createElement({ ...basicProps, modelSelectorId, categorySelectorId } as ElementProps), IModelError); // Missing displayStyleId
```
Comment (kabentley) [inline core/backend/src/test/standalone/ViewDefinition.test.ts:70]: add expected error message

**Review by kabentley — CHANGES_REQUESTED** [review]

File: core/backend/src/test/standalone/ViewDefinition.test.ts (hunk 3 above)
Comment (ramanujam-raman) [reply core/backend/src/test/standalone/ViewDefinition.test.ts]: Fixed.

File: core/backend/src/test/standalone/ViewDefinition.test.ts (hunk 4 above)
Comment (ramanujam-raman) [reply core/backend/src/test/standalone/ViewDefinition.test.ts]: Fixed.

File: core/backend/src/test/standalone/ViewDefinition.test.ts (hunk 5 above)
Comment (ramanujam-raman) [reply core/backend/src/test/standalone/ViewDefinition.test.ts]: Fixed.

File: core/backend/src/test/integration/HubUtility.ts (hunk 1 above)
Comment (kabentley) [reply core/backend/src/test/integration/HubUtility.ts]: i removed this stuff

**Review by kabentley — APPROVED** [review]

**Pull request merged**